src/app/pages/LiquidityMining/components/HistoryTable/index.tsx

```diff
@@ -1,23 +1,84 @@
-import React from 'react';
+import React, { useEffect, useState, useCallback } from 'react';
+import axios from 'axios';
+
+import { backendUrl, currentChainId } from 'utils/classifiers';
+import { useAccount } from 'app/hooks/useAccount';
 import { TableBody } from './TableBody';
 import { TableHeader } from './TableHeader';
+import { Pagination } from 'app/components/Pagination';
+
+const pageSize = 6;
 
 export const HistoryTable: React.FC = () => {
+  const account = useAccount();
+  const url = backendUrl[currentChainId];
+  const [history, setHistory] = useState([]) as any;
+  const [loading, setLoading] = useState(false);
+  const [page, setPage] = useState(1);
+  const [total, setTotal] = useState(0);
+
+  let cancelTokenSource;
+  const getData = () => {
+    if (cancelTokenSource) {
+      cancelTokenSource.cancel();
+    }
+
+    cancelTokenSource = axios.CancelToken.source();
+    axios
+      .get(
+        `${url}/liquidity-history/${account}?page=${page}&pageSize=${pageSize}`,
+        {
+          cancelToken: cancelTokenSource.token,
+        },
+      )
+      .then(res => {
+        const { events, pagination } = res.data;
+        setHistory(events || []);
+        setTotal(pagination.totalPages * pageSize);
+        setLoading(false);
+      })
+      .catch(e => {
+        console.log(e);
+        setHistory([]);
+        setLoading(false);
+      });
+  };
```

Comment:
I added `useFetch` hook some time ago, maybe it would be useful in cases like this

Reply:
Well since it's now using new api model and pagination... useFetch needs to be changed to support this.